Loop/View Controllers/SettingsTableViewController.swift

```diff
@@ -378,100 +384,21 @@ final class SettingsTableViewController: UITableViewController, IdentifiableClas
-                            tableView.deselectRow(at: indexPath, animated: true)
-                        }
-
-                        present(alert, animated: true, completion: nil)
-                    default:
-                        break
-                    }
                 }
             }
         case .cgm:
-            if let cgmManager = dataManager.cgmManager as? CGMManagerUI {
-                if let unit = dataManager.loopManager.glucoseStore.preferredUnit {
-                    var settings = cgmManager.settingsViewController(for: unit)
-                    settings.completionDelegate = self
-                    present(settings, animated: true)
-                    tableView.deselectRow(at: indexPath, animated: true)
-                }
-            } else if dataManager.cgmManager is PumpManagerUI {
-                // The pump manager is providing glucose, but allow reverting the CGM
-                let alert = UIAlertController(deleteCGMManagerHandler: { [weak self] (isDeleted) in
-                    if isDeleted {
-                        self?.dataManager.cgmManager = nil
-                    }
-
-                    tableView.deselectRow(at: indexPath, animated: true)
-                    self?.updateCGMManagerRows()
-                })
-                present(alert, animated: true, completion: nil)
-            } else {
-                // Add new CGM
-                let cgmManagers = dataManager.availableCGMManagers
-
-                switch cgmManagers.count {
-                case 1:
-                    if let cgmManager = cgmManagers.first, let CGMManagerType = dataManager.cgmManagerTypeByIdentifier(cgmManager.identifier) {
-                        setupCGMManager(CGMManagerType)
-                    }
-
-                    tableView.deselectRow(at: indexPath, animated: true)
-                case let x where x > 1:
-                    let alert = UIAlertController(cgmManagers: cgmManagers, pumpManager: dataManager.pumpManager as? CGMManager) { [weak self] (identifier, pumpManager) in
-                        if let self = self {
-                            if let cgmManagerIdentifier = identifier, let CGMManagerType = self.dataManager.cgmManagerTypeByIdentifier(cgmManagerIdentifier) {
-                                self.setupCGMManager(CGMManagerType)
-                            } else if let pumpManager = pumpManager {
-                                self.completeCGMManagerSetup(pumpManager)
-                            }
-                        }
-
-                        tableView.deselectRow(at: indexPath, animated: true)
-                    }
-
-                    alert.addCancelAction { (_) in
-                        tableView.deselectRow(at: indexPath, animated: true)
-                    }
-
-                    present(alert, animated: true, completion: nil)
-                default:
-                    break
-                }
+            cgmSelected {
```

Comment:
This code was copied as-is to a new function so that it could be re-used by the new UI ViewModel.